modules/messages/src/lib.rs

```diff
@@ -1962,4 +2066,37 @@ mod tests {
 		MessagesOperatingMode::Basic(BasicOperatingMode::Normal),
 		MessagesOperatingMode::Basic(BasicOperatingMode::Halted)
 	);
+
+	#[test]
+	fn inbound_storage_extra_proof_size_bytes_works() {
+		fn relayer_entry() -> UnrewardedRelayer<TestRelayer> {
+			UnrewardedRelayer { relayer: 42u64, messages: DeliveredMessages { begin: 0, end: 100 } }
+		}
+
+		fn storage(relayer_entries: usize) -> RuntimeInboundLaneStorage<TestRuntime, ()> {
+			RuntimeInboundLaneStorage {
+				lane_id: Default::default(),
+				cached_data: RefCell::new(Some(InboundLaneData {
+					relayers: vec![relayer_entry(); relayer_entries].into_iter().collect(),
+					last_confirmed_nonce: 0,
+				})),
+				_phantom: Default::default(),
+			}
+		}
+
+		let max_entries = crate::mock::MaxUnrewardedRelayerEntriesAtInboundLane::get() as usize;
+
+		// when we have exactly `MaxUnrewardedRelayerEntriesAtInboundLane` unrewarded relayers
+		assert_eq!(storage(max_entries).extra_proof_size_bytes(), 0);
+
+		// when we have less than `MaxUnrewardedRelayerEntriesAtInboundLane` unrewarded relayers
+		assert_eq!(
+			storage(max_entries - 1).extra_proof_size_bytes(),
+			relayer_entry().encode().len() as u64
+		);
```

Comment:
nit1: I hope `encoded_size_hint()` used in `extra_proof_size_bytes()` is consistent in its return value and this test will be robust - worst case it will intermittently fail and we'll worry about that then.

nit2: I would add another one for extra safety (especially since previous test only verifies generic `post_dispatch_weight.proof_size() < pre_dispatch_weight.proof_size()` without checking/knowing how much smaller it should be.

```suggestion
		assert_eq!(
			storage(max_entries - 1).extra_proof_size_bytes(),
			relayer_entry().encode().len() as u64
		);
		assert_eq!(
			storage(max_entries - 2).extra_proof_size_bytes(),
			2 * relayer_entry().encode().len() as u64
		);
```
(above could also be a loop, but testing at least two values is good enough imo)

Reply:
> nit1: I hope encoded_size_hint() used in extra_proof_size_bytes() is consistent in its return value and this test will be robust - worst case it will intermittently fail and we'll worry about that then.

I'm not sure exactly what you mean here, but both expressions in the `extra_proof_size_bytes` are using values, returned from the `max_encoded_len`. So iiuc they are somewhat consistent :)

> nit2: I would add another one for extra safety (especially since previous test only verifies generic post_dispatch_weight.proof_size() < pre_dispatch_weight.proof_size() without checking/knowing how much smaller it should be.

Done